packages/client/src/utils/firestore/compat.ts

```diff
@@ -94,12 +94,14 @@ export const doc = (
   return match(node, {
     [FirestoreEnv.Client]: ({ instance }) =>
       FirestoreDocVariant.client({
-        // Even though 'doc' (aliased as 'clientDoc' here) works on both firestore and collection instanceerences,
+        // Even though 'doc' (aliased as 'clientDoc' here) works on both firestore and collection references,
         // it doesn't work well on unions because of slightly different type overloads, hence the typecast.
-        instance: clientDoc(
-          instance as ClientFirestore,
-          pathSegments.join("/")
-        ),
+        instance: docPath.length
+          ? clientDoc(
+              instance as ClientCollectionReference,
+              docPath.length ? docPath : undefined
```

Comment:
Isn't this condition redundant? We're in a conditional branch where `docPath.length` is `true`. I'd replace `docPath.length ? docPath : undefined` with `docPath`.

Reply:
Yup, good catch. This is a leftover from previous attempt (using only `clientDoc(instance, docPath.length ? docPath : undefined)`, without the first check) which didn't seem to work